generative_ai/prompts/prompt_template.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+PROJECT_ID = os.getenv("GOOGLE_CLOUD_PROJECT")
+
+
+def prompt_template_example() -> list:
+    """Build a parameterized prompt template to generate content with multiple variable sets"""
+
+    # [START generativeaionvertexai_prompt_template]
+    import vertexai
+    from vertexai.preview.prompts import Prompt
+
+    # Initialize vertexai
+    vertexai.init(project=PROJECT_ID, location="us-central1")
+
+    variables = [
+        {"animal": "Eagles", "activity": "eat berries"},
+        {"animal": "Coyotes", "activity": "jump"},
+        {"animal": "Squirrels", "activity": "fly"}
+    ]
+
+    # define prompt template
+    prompt = Prompt(
+        prompt_data="Do {animal} {activity}?",
+        model_name="gemini-1.5-flash-002",
+        variables=variables,
+        system_instruction=["You are a helpful zoologist"]
```

Comment:
Brackets aren't necessary, you can just input the system instruction as a string